Source/Classes/BicepRestoreParametersRelativePath.ps1

```diff
@@ -0,0 +1,45 @@
+class BicepRestoreParametersRelativePath : BicepRestoreParametersBase {
+    
+    BicepRestoreParametersRelativePath([object]$Object) {
+        BicepRestoreParametersRelativePath($Object, $PWD.Path)
+    }
+
+    BicepRestoreParametersRelativePath([string]$Fullname) {
+        $RelativePath = Resolve-Path -Path $Fullname -Relative -ErrorAction 'Stop'
+        $SplitPath = $RelativePath.Split([System.IO.Path]::DirectorySeparatorChar)
+        $FileNamePattern = '^(.+)-(\d+)\.bicep$'
+        $This.Name = $SplitPath[-1] -replace $FileNamePattern, '$1'
+        $This.Version = '{0}-{1}' -f $SplitPath[-2], ($SplitPath[-1] -replace $FileNamePattern, '$2')
+        $This.Type = 'RelativePath'
+        $This.Source = $SplitPath[ - $SplitPath.Length..-3] -join '/'
+    }
+    
+    BicepRestoreParametersRelativePath([object]$Object, [string]$BasePath) {
+        if ($Object.Version -notmatch '^\d\d\d\d-\d\d-\d\d-\d$') {
+            throw ('{0} is not a valid version.' -f $Object.Version)
+        }
+        Push-Location -Path $BasePath -ErrorAction 'Stop'
+        $this.Type = $Object.Type 
+        $this.Version = $Object.Version 
+        $this.Name = $Object.Name 
+        $this.Source = Resolve-Path -Path $Object.Source -ErrorAction 'Stop'
+        $this.BasePath = $BasePath
+        Pop-Location -ErrorAction 'Stop'
+    }
+
+    [void] Restore($Path) {
+        $DestinationPath = $this.ResolveDestinationPath($Path)
+        
+        $ApiVersion = $this.Version -replace '-\d+$'
+        $ModuleVersion = $this.Version -replace '.+-(\d+)$', '$1'
+        $SourceFileName = '{0}-{1}.bicep' -f $this.Name, $ModuleVersion
+        $SourcePath = Join-Path -Path $this.Source -ChildPath $APIVersion -AdditionalChildPath $SourceFileName
+        
+        if (-not (Test-Path -Path $SourcePath -PathType 'Leaf')) {
+            throw "File not found: $SourcePath"
```

Comment:
Not sure if I like the throw when a file in the dependencies.json can't be found. If I have multiple resource modules in the dependencies.json and I've made manual changes to the dependencies.json file (like changed version) or moved/renamed bicep modules, I might end up re-running Restore-Bicep multiple times to fix all errors because it stops for every error. I think I want the cmdlet to write the error to the terminal but continue to process the rest of the dependencies.